Source/Core/Core/HW/SI/SI_DeviceAMBaseboard.cpp

```diff
@@ -0,0 +1,2361 @@
+              }
+
+              DEBUG_LOG_FMT(SERIALINTERFACE_JVSIO, "JVS-IO: Command 0x25, PositionInput:{}",
+                            channel);
+              break;
+            }
+            case JVSIOCommand::CoinSubOutput:
+            {
+              u32 slot = *jvs_io++;
+              m_coin[slot] -= (*jvs_io++ << 8) | *jvs_io++;
+              message.addData(StatusOkay);
+              DEBUG_LOG_FMT(SERIALINTERFACE_JVSIO, "JVS-IO: Command 0x30, CoinSubOutput: {}", slot);
+              break;
+            }
+            case JVSIOCommand::GeneralDriverOutput:
+            {
+              u32 bytes = *jvs_io++;
+
+              if (bytes)
+              {
+                message.addData(StatusOkay);
+
+                // The lamps are controlled via this
+                if (AMMediaboard::GetGameType() == MarioKartGP)
+                {
+                  u32 status = *jvs_io++;
+                  if (status & 4)
+                  {
+                    DEBUG_LOG_FMT(SERIALINTERFACE_JVSIO, "JVS-IO: Command 32, Item Button ON");
+                  }
+                  else
+                  {
+                    DEBUG_LOG_FMT(SERIALINTERFACE_JVSIO, "JVS-IO: Command 32, Item Button OFF");
+                  }
+                  if (status & 8)
+                  {
+                    DEBUG_LOG_FMT(SERIALINTERFACE_JVSIO, "JVS-IO: Command 32, Cancel Button ON");
+                  }
+                  else
+                  {
+                    DEBUG_LOG_FMT(SERIALINTERFACE_JVSIO, "JVS-IO: Command 32, Cancel Button OFF");
+                  }
+                  break;
+                }
+
+                Common::UniqueBuffer<u8> buf(bytes);
+
+                for (u32 i = 0; i < bytes; ++i)
+                {
+                  buf[i] = *jvs_io++;
+                }
+
+                INFO_LOG_FMT(
+                    SERIALINTERFACE_JVSIO,
+                    "JVS-IO: Command 0x32, GPO: {:02x} {:02x} {} {:02x}{:02x}{:02x} ({:02x})",
+                    delay, m_rx_reply, bytes, buf[0], buf[1], buf[2],
+                    Common::swap16(*reinterpret_cast<u16*>(&buf[1])) >> 2);
+
+                /*
+                  Handling of the motion seat used in F-Zero AXs DX version
+                */
+                switch (Common::swap16(*reinterpret_cast<u16*>(&buf[1])) >> 2)
+                {
+                case 0x70:
+                  delay++;
+                  if ((delay % 10) == 0)
+                  {
+                    m_rx_reply = 0xFB;
+                  }
+                  break;
+                case 0xF0:
+                  m_rx_reply = 0xF0;
+                  break;
+                default:
+                case 0xA0:
+                case 0x60:
+                  break;
+                }
+              }
+              break;
+            }
+            case JVSIOCommand::CoinAddOutput:
+            {
+              int slot = *jvs_io++;
+              m_coin[slot] += (*jvs_io++ << 8) | *jvs_io++;
```

Comment:
The order of evaluation of the `*jvs_io++` are unspecified and might not be what you want.

This produces a `-Wunsequenced` warning.